src/WorldStorage/FastNBT.cpp

```diff
@@ -91,9 +91,9 @@ bool cParsedNBT::ReadString(size_t & a_StringStart, size_t & a_StringLen)
 	NEEDBYTES(2);
 	a_StringStart = m_Pos + 2;
 	a_StringLen = static_cast<size_t>(GetBEShort(m_Data + m_Pos));
-	if (a_StringLen > 0xffff)
+	if (a_StringLen > m_Length - m_Pos - 2)
```

Comment:
Can't you just change it to `if (a_StringLen + 2 > m_Length - m_Pos)` ? It's clean and it's robust.